doc/tutorials/install-config-files.md

```diff
@@ -32,25 +31,25 @@ You have to enter the paths accordingly.
 Read the tutorials on mounting and namespaces if you are not sure what this means.
 
 ```sh
-echo -e "keyA=a\nkeyB=b\nkeyC=Y" > ~/installing.ini
-kdb install-config-file user/tests/installing /installing.ini ini
+mkdir -p /tmp/new
+echo -e "keyA=a\nkeyB=b\nkeyC=Y" > /tmp/new/installing.ini
+kdb install-config-file system/tests/installing /tmp/new/installing.ini ini
 ```
 
 We can check that this worked by calling
 
 ```sh
-kdb get user/tests/installing/keyB
+kdb get system/tests/installing/keyB
 #> X
-kdb get user/tests/installing/keyC
+kdb get system/tests/installing/keyC
 #> Y
 ```
 
 Finally, we use the following commands to clean up our tutorial.
 
 ```sh
-kdb umount user/tests/installing
-kdb rm -rf user/elektra/merge/preserve
-rm ~/installing.ini
-rm ~/.config/installing.ini
-rmdir --ignore-fail-on-non-empty ~/.config
+kdb umount system/tests/installing
+kdb rm -rf user/elektra/merge/preserve/installing.ini
+rm /tmp/installing.ini
```

Comment:
@sanssecours do we have somewhere a description how to use tmp files in shell recorder?

Reply:
Good question! I didn't get this working yet with files that I had in `~` and `~/.config` either. There were problems like one time `~/.config` exists, one time you have to delete it...

It seems like currently my `kdb mount` and `kdb ls` don't work. Is there a way to display more (debug) information than only the exit code?

Reply:
> do we have somewhere a description how to use tmp files in shell recorder?

Not that I know of. If you want to create temporary file, it might make sense to use the command [`mktemp`](https://manpage.me/?q=mktemp) and store the path to the temporary file in the KDB. Normally you would just use a (shell) variable for that purpose. However, the Markdown Shell Recorder does currently not support variables.

The command below shows how we use `mktemp` in the [Markdown Shell Recorder test of the `path` plugin](https://github.com/ElektraInitiative/libelektra/tree/master/src/plugins/path).

```sh
kdb set user/tests/path/tempfile $(mktemp)
```

> It seems like currently my `kdb mount` and `kdb ls` don't work. Is there a way to display more (debug) information than only the exit code?

The Shell Recorder should print a ”protocol” including the output to `stderr` and `stdout` if the test failed. For example in the [build job `🐧 FULL` the command `kdb install-config-file` failed](https://travis-ci.org/ElektraInitiative/libelektra/jobs/608044028#L2069-L2073), because the Shell Recorder was unable to locate the command `kdb`:

```
CMD: kdb install-config-file system/tests/installing /tmp/new/installing.ini ini
RET: 30
=== FAILED return value does not match expected pattern 0
STDERR: /home/travis/build/ElektraInitiative/libelektra/install/lib/elektra/tool_exec/install-config-file: 35: /home/travis/build/ElektraInitiative/libelektra/install/lib/elektra/tool_exec/install-config-file: kdb: not found
ERROR: Could not use kdb ls on system/tests/installing.
```

. This command does not exist if `BUILD_SHARED` is not enabled. Instead you should be able to use the command `kdb-full` (`BUILD_FULL=ON`) or `kdb-static` (`BUILD_STATIC=ON`).

Reply:
Thank you, @sanssecours !

> The Shell Recorder should print a ”protocol” including the output to stderr and stdout if the test failed.

The problem is that sometimes the protocol part is missing. 
In this case we only get
```
kdb install-config-file system/tests/installing $(kdb get user/tests/tempfiles/secondFile) ini

ERROR - RET:
Return value “35” does not match “0”

kdb install-config-file system/tests/installing `kdb get user/tests/tempfiles/firstFile` ini

ERROR - RET:
Return value “34” does not match “0”
```

You can see such a full log output [here](https://pastebin.com/zN3uyxqy) as GitHub file upload does not seem to work for me currently.

> This command does not exist if BUILD_SHARED is not enabled. Instead you should be able to use the command kdb-full (BUILD_FULL=ON) or kdb-static (BUILD_STATIC=ON).

Thank you! I would have looked for that for years.

> If you want to create temporary file, it might make sense to use the command mktemp and store the path to the temporary file in the KDB. 

I agree that `mktemp` is a nice solution if we want to use temporary files and have pushed 110e1ce9852eda5e20530bdb181f2d320cc18f28 which implements this.


I assume the danger that @markus2330 meant [here](https://github.com/ElektraInitiative/libelektra/pull/3106#discussion_r342669104) is that a user could have a file with exactly the name `installing.ini` in his home directory and accidentally delete it. I see that danger, too and have pushed the changes according to the feedback.

However, I actually disagree that it's better to use a temporary file for the tutorial. I assume that the chance that someone already uses exactly this file and is so incautious that he or she accidentally deletes it is pretty low. In contrast, we have to use such a monstrosity
```sh
kdb set user/tests/tempfiles/secondFile $(echo $(mktemp -d --tmpdir dir-for-file-XXXXX)/$(basename $(kdb get user/tests/tempfiles/firstFile)))
```
to avoid this risk and this makes the tutorial disproportionally more difficult to comprehend, in my opinion. Additionally, something like changing an actually used hosts file (as we do [here](https://github.com/ElektraInitiative/libelektra/blob/master/doc/tutorials/mount.md)) seems much more dangerous to me.

Reply:
> The problem is that sometimes the protocol part is missing.

In theory you can use the switch `-p` to always print the protocol. However, that option is not really exposed via `add_msr_test`. I opened PR #3175, which changes the behavior of the Shell Recorder. If we apply the code updates of the pull request, then the Shell Recorder should always print the protocol for a failed test.

Reply:
@sanssecours Did I miss something or is `mktemp` really not POSIX compliant? I find it neither in the [index](https://pubs.opengroup.org/onlinepubs/9699919799/idx/im.html) nor in the [table of contents](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/contents.html). Our bashism checks doesn't complain, however.

Reply:
It helps to look what is already done in other scripts, as this is known to pass the build server.

E.g. `doc/tutorials/validation.md:kdb set system/tests/specbackup $(mktemp)`

Reply:
> Did I miss something or is `mktemp` really not POSIX compliant?

Nope. I think you are right, the `mktemp` shell command is not defined  by POSIX. However, the command should be available on most Unix like operating systems.